tests/test_completion.py

```diff
@@ -113,6 +113,23 @@ def test_complete_bogus_command(cmd2_app):
     first_match = complete_tester(text, line, begidx, endidx, cmd2_app)
     assert first_match is None
 
+def test_complete_macro(base_app, request):
+    # Create the macro
+    out = run_cmd(base_app, 'macro create fake pyscript {1}')
+    assert out == normalize("Macro 'fake' created")
+
+    # Macros do path completion
+    test_dir = os.path.dirname(request.module.__file__)
+
+    text = os.path.join(test_dir, 'script.py')
```

Comment:
If I'm understanding this right, this unit test assumes script.py is in the test directory and verifies that if the user typed a macro name, then a complete filename, then pressed tab, the tab completion would append a space. If that's right, it seems underwhelming. Why not show that it can actually complete a filename?

You've done a ton of work here and I don't want to seem ungrateful, though. I hope I'm misinterpreting the code or there's a good reason you've done it this way.